tiledb/sm/query/readers/aggregators/count_aggregator.cc

```diff
@@ -0,0 +1,126 @@
+/**
+ * @file count_aggregator.cc
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2023 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * This file implements class CountAggregator.
+ */
+
+#include "tiledb/sm/query/readers/aggregators/count_aggregator.h"
+
+#include "tiledb/sm/query/query_buffer.h"
+#include "tiledb/sm/query/readers/aggregators/aggregate_buffer.h"
+
+using namespace tiledb::common;
+
+namespace tiledb {
+namespace sm {
+
+class CountAggregatorStatusException : public StatusException {
+ public:
+  explicit CountAggregatorStatusException(const std::string& message)
+      : StatusException("CountAggregator", message) {
+  }
+};
+
+CountAggregator::CountAggregator(const std::string field_name)
+    : count_(0) {
+  if (field_name != constants::all_attributes) {
+    throw CountAggregatorStatusException(
+        "Count aggregates must be requested for all attributes only.");
```

Comment:
I assume this is a temporary while the PR is still in draft? Seems odd that we have to request counts for all fields.

Reply:
No, this is meant to be permanent. This is implementing the aggregator for count that would be used in this SQL query: SELECT COUNT(*) WHERE dim1 > 10. Maybe the error message is what is tripping you up?